### DISCUSSION TITLE
yarn install failed due to electron

### DISCUSSION THREAD

**Discussion by henryhe1993** [eclipse-theia/theia#9447] [Q&A] [answered]
I am trying the quick start of Theia browser example step by step. There is an error when i run yarn install. 
MacOS v11.2.2(m1 chip running Rosetta).
node -v v12.14.1
yarn -v 1.22.10

Here is the error:
yarn install v1.22.10
$ node-gyp install

....
```
Exit code: 1
Command: node install.js
Arguments: 
Directory: /Users/hetianlun/Public/practice/theia/node_modules/electron
Output:
HTTPError: Response code 405 (Method Not Allowed)
    at EventEmitter.<anonymous> (/Users/hetianlun/Public/practice/theia/node_modules/@electron/get/node_modules/got/source/as-stream.js:35:24)
    at EventEmitter.emit (events.js:223:5)
    at module.exports (/Users/hetianlun/Public/practice/theia/node_modules/@electron/get/node_modules/got/source/get-response.js:22:10)
    at ClientRequest.handleResponse (/Users/hetianlun/Public/practice/theia/node_modules/@electron/get/node_modules/got/source/request-as-event-emitter.js:155:5)
    at Object.onceWrapper (events.js:313:26)
    at ClientRequest.emit (events.js:228:7)
    at ClientRequest.origin.emit (/Users/hetianlun/Public/practice/theia/node_modules/@szmarczak/http-timer/source/index.js:37:11)
```

Not sure why there is a http method not allowed error. I have never met this error before using yarn/npm.

```
yarn config list
yarn config v1.22.10
info yarn config
{
  'version-tag-prefix': 'v',
  'version-git-tag': true,
  'version-commit-hooks': true,
  'version-git-sign': false,
  'version-git-message': 'v%s',
  'init-version': '1.0.0',
  'init-license': 'MIT',
  'save-prefix': '^',
  'bin-links': true,
  'ignore-scripts': false,
  'ignore-optional': false,
  registry: 'https://registry.yarnpkg.com',
  'strict-ssl': true,
  'user-agent': 'yarn/1.22.10 npm/? node/v12.14.1 darwin x64',
  '@ali:registry': 'http://registry.npm.alibaba-inc.com',
  '@alife:registry': 'https://registry.npm.alibaba-inc.com',
  electron_mirror: 'https://registry.npmjs.org/',
  lastUpdateCheck: 1620374271225,
  sass_binary_site: 'https://tnpm-hz.oss-cn-hangzhou.aliyuncs.com/dist/node-sass',
  'unsafe-disable-integrity-migration': true
}
info npm config
{
  email: '', 
registry: 'http://registry.npmjs.org'
}
```

Or if I am only interested in browser version, is it safe to completely remove electron from dependencies? Any help is much appreciated.

**Comment by vince-fugnitto** [comment, marked as answer]
> Or if I am only interested in browser version, is it safe to completely remove electron from dependencies? Any help is much appreciated.

@henryhe1993 thank you for the discussion, I don't see your application `package.json` so I assume you have `electron` as a dependency despite targeting the browser? If so, you can remove `@theia/electron` as it is not needed for the browser and will likely cause issues. You can take a look at the `example-browser` example present in this repository:

https://github.com/eclipse-theia/theia/blob/master/examples/browser/package.json

**Reply by henryhe1993** [reply]
Thank you for your reply. 

I ran "yarn install" under the project root folder so it did include @theia/electron. 

Another question, i see there are some node_modules are prefixed by "_" , e.g. "_ansi-colors@x.x.x@ansi-colors", and there is an equivalent normal package "ansi-colors@x.x.x". I tried to google "the meaning of _", but I hardly found any useful information. Would you please give me any clue to that. or any link is appreciated.

**Reply by vince-fugnitto** [reply]
@henryhe1993 I do not see the use of `_` in any of the framework dependencies, do you have a concrete example?
Verifying the [documentation of npm](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#name), the underscore `_` is only allowed for `scoped` packages:
> The names of scoped packages can begin with a dot or an underscore. This is not permitted without a scope.

**Reply by henryhe1993** [reply]
I am sorry, I cannot reproduce it now. It seems those packages will disappear once I close my editor (vscode). So I guess those were some temporary packages after a failed installation which were cached somehow. 
This question was just for my curiosity to check if I missed something about npm.  Anyway, thanks a lot for your information.
